transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/ListTransferDestinationViewModel.java

```diff
@@ -0,0 +1,84 @@
+package com.hyperwallet.android.ui.transfer.viewmodel;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.hyperwallet.android.model.HyperwalletErrors;
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.ui.common.repository.Event;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepository;
+
+import java.util.List;
+
+public class ListTransferDestinationViewModel extends ViewModel {
+
+    private final MutableLiveData<List<HyperwalletTransferMethod>> mTransferDestinationList = new MutableLiveData<>();
+    private final MutableLiveData<Event<HyperwalletTransferMethod>> mTransferDestinationSection =
+            new MutableLiveData<>();
+    private final MutableLiveData<Event<HyperwalletErrors>> mTransferDestinationError = new MutableLiveData<>();
+    private TransferMethodRepository mTransferMethodRepository;
+
+    public ListTransferDestinationViewModel(
+            @NonNull final TransferMethodRepository transferMethodRepository) {
+        mTransferMethodRepository = transferMethodRepository;
+        loadTransferDestinationList();
+    }
+
+    public void selectTransferDestination(HyperwalletTransferMethod transferMethod) {
+        mTransferDestinationSection.postValue(new Event<>(transferMethod));
+    }
+
+    private void loadTransferDestinationList() {
+        mTransferMethodRepository.loadTransferMethods(new TransferMethodRepository.LoadTransferMethodListCallback() {
+            @Override
+            public void onTransferMethodListLoaded(List<HyperwalletTransferMethod> transferMethods) {
+                mTransferDestinationList.postValue(transferMethods);
+            }
+
+            @Override
+            public void onError(HyperwalletErrors errors) {
+                mTransferDestinationError.postValue(new Event<>(errors));
+            }
+        });
+    }
+
+    public MutableLiveData<List<HyperwalletTransferMethod>> getTransferDestinationList() {
+        return mTransferDestinationList;
+    }
+
+    public MutableLiveData<Event<HyperwalletTransferMethod>> getTransferDestinationSection() {
+        return mTransferDestinationSection;
+    }
+
+    /**
+     * @see ViewModel#onCleared()
+     */
+    @Override
+    protected void onCleared() {
+        super.onCleared();
+        mTransferMethodRepository = null;
```

Comment:
we don't need this

Reply:
fixed here https://github.com/hyperwallet/hyperwallet-android-ui-sdk/pull/79/commits/54cb98c44152d4dfff46cfa4b2947a221645dacf